src/server/migrations/migrateDatabase.js

```
@@ -0,0 +1,191 @@
+		path[vertex] = -1;
+	}
+
+	queue.push(curr);
+	visited[curr] = true;
+
+	while (queue.length > 0) {
+		const currentVertexID = queue.shift();
+		const currentVertex = adjListArray[currentVertexID];
+		const edges = currentVertex.length;
+		for (let i = 0; i < edges; i++) {
+			const target = currentVertex[i];
+			if (!visited[target.toVersion]) {
+				visited[target.toVersion] = true;
+				path[target.toVersion] = currentVertexID;
+				queue.push(target.toVersion);
+			}
+		}
+	}
+	return path;
+}
+
+/**
+ * Based on the path array, recursively find the correct file to the version wanted to update
+ * and add it to the requiredFile array
+ * @param curr current version of the database
+ * @param to version want to migrate to
+ * @param path array that store the indexes to the version that we want to migrate to
+ */
+function getRequiredFilesToMigrate(curr, to, path) {
+	if (curr === to) {
+		requiredFile.push();
+	} else if (path[to] === -1) {
+		throw new Error('No path found');
+	} else {
+		getRequiredFilesToMigrate(curr, path[to], path);
+		requiredFile.push({
+			fromVersion: path[to],
+			toVersion: to
+		});
+	}
+}
+
+/**
+ * Open a database transaction and migrate the database by calling up() method.
+ * Insert row into migration folder
+ * @param neededFiles name of files needed to migrate
+ * @param allMigrationFiles is all of possible migration files
+ */
+async function migrateDatabaseTransaction(neededFiles, allMigrationFiles) {
+	try {
+		await db.tx(async t => {
+			neededFiles.forEach(neededFile => {
+				allMigrationFiles.forEach(async migrationFile => {
+					if (neededFile.fromVersion === migrationFile.fromVersion && neededFile.toVersion === migrationFile.toVersion) {
+						try {
+							await migrationFile.up(t);
+							const migration = new Migration(undefined, migrationFile.fromVersion, migrationFile.toVersion);
+							await migration.insert(t);
+						} catch (err) {
+							log.error('Error while migrating database', err);
```

You should throw an error here. You want to abort the transaction if something goes wrong.